WordPressAuthenticatorTests/Authenticator/WordPressSourceTagTests.swift

```diff
@@ -0,0 +1,175 @@
+    
+    func testLoginWPComPasswordSourceTag() {
+        let tag = WordPressSupportSourceTag.loginWPComPassword
+        
+        let nameExpectation = "loginWPComPassword"
+        let originExpectation = "origin:login-wpcom-password"
+        
+        XCTAssertEqual(tag.name, nameExpectation)
+        XCTAssertEqual(tag.origin, originExpectation)
+    }
+    
+    func testWPComSignupEmailSourceTag() {
+        let tag = WordPressSupportSourceTag.wpComSignupEmail
+        
+        let nameExpectation = "wpComSignupEmail"
+        let originExpectation = "origin:wpcom-signup-email-entry"
+        
+        XCTAssertEqual(tag.name, nameExpectation)
+        XCTAssertEqual(tag.origin, originExpectation)
+    }
+    
+    func testWPComSignupSourceTag() {
+        let tag = WordPressSupportSourceTag.wpComSignup
+        
+        let nameExpectation = "wpComSignup"
+        let originExpectation = "origin:signup-screen"
+        
+        XCTAssertEqual(tag.name, nameExpectation)
+        XCTAssertEqual(tag.origin, originExpectation)
+    }
+    
+    func testWPComSignupWaitingForGoogleSourceTag() {
+        let tag = WordPressSupportSourceTag.wpComSignupWaitingForGoogle
+        
+        let nameExpectation = "wpComSignupWaitingForGoogle"
+        let originExpectation = "origin:signup-waiting-for-google"
+        
+        XCTAssertEqual(tag.name, nameExpectation)
+        XCTAssertEqual(tag.origin, originExpectation)
+    }
+    
+    func testWPComAuthGoogleSignupWaitingForGoogleSourceTag() {
+        let tag = WordPressSupportSourceTag.wpComAuthWaitingForGoogle
+        
+        let nameExpectation = "wpComAuthWaitingForGoogle"
+        let originExpectation = "origin:auth-waiting-for-google"
+        
+        XCTAssertEqual(tag.name, nameExpectation)
+        XCTAssertEqual(tag.origin, originExpectation)
+    }
+    
+    func testWPComAuthGoogleSignupConfirmationSourceTag() {
+        let tag = WordPressSupportSourceTag.wpComAuthGoogleSignupConfirmation
+        
+        let nameExpectation = "wpComAuthGoogleSignupConfirmation"
+        let originExpectation = "origin:auth-google-signup-confirmation"
+        
+        XCTAssertEqual(tag.name, nameExpectation)
+        XCTAssertEqual(tag.origin, originExpectation)
+    }
+    
+    func testWPComSignupMagicLinkSourceTag() {
+        let tag = WordPressSupportSourceTag.wpComSignupMagicLink
+        
+        let nameExpectation = "wpComSignupMagicLink"
+        let originExpectation = "origin:signup-magic-link"
+        
+        XCTAssertEqual(tag.name, nameExpectation)
+        XCTAssertEqual(tag.origin, originExpectation)
+    }
+    
+    func testWPComSignupAppleSourceTag() {
+        let tag = WordPressSupportSourceTag.wpComSignupApple
+        
+        let nameExpectation = "wpComSignupApple"
+        let originExpectation = "origin:signup-apple"
+        
+        XCTAssertEqual(tag.name, nameExpectation)
+        XCTAssertEqual(tag.origin, originExpectation)
+    }
+}
```

Comment:
I'd normally argue against writing tests for constant value, like here were the tests checks that each `static var` on `WordPressSupportSourceTag` calls the init method with the expected values. There is basically no logic in here to test.

But, in this case the test might be useful. Since the tag values are shared with external systems, it's valuable to have an extra check to prevent devs from inadvertently change them.